src/plugins/unified_search/public/filter_bar/filter_item/filter_item.tsx

```diff
@@ -342,6 +353,37 @@ export function FilterItem(props: FilterItemProps) {
     return label;
   }
 
+  function getFilterString() {
```

Comment:
I had to move this logic up one level from `FilterView` up to `FilterItem` in order to accommodate using an `EuiToolTip` instead of the native browser tooltip that was used before.

Basically, my original approach was to simply wrap the `EuiToolTip` around the `EuiBadge` in `FilterView` like so:

https://github.com/elastic/kibana/blob/2a49e93cec0c67d8ecdafacec3ec91245b01cbec/src/plugins/unified_search/public/filter_bar/filter_view/index.tsx#L97-L111

However, this caused an issue where the filter pills in the top navigation bar would no longer be truncated - if the screen size was small enough and the filter was large enough, it would simply go off the edge of the screen without truncation. So, to get around this, I had to move all the `filterString/toolTip` logic up and wrap it around the entire component - with this, truncation occurs in regardless of whether or not `readOnly` is `true` or `false`.